lib/src/http/auto_retry.dart

```diff
@@ -170,18 +171,60 @@ class RetryBackOff {
   /// The back-off method to use. One of none, fixed, or exponential.
   final RetryBackOffMethod method;
 
+  /// Whether to enable jitter or not.
+  final bool enableJitter;
+
+  /// The maximum duration between retries.
+  final maxDurationInMs;
+
+  /// The default maximum duration between retries. (5 minutes)
+  static const defaultMaxDurationInMs = 5 * 60 * 1000;
+
+  int _baseBackOffFunction(int numAttempts) {
+    return this.duration.inMilliseconds * pow(2, numAttempts);
+  }
+
+  Duration calculate(int numAttempts) {
```

Comment:
I don't think this should be a part of the `RetryBackOff` class. This class is meant to be used by consumers to create representations of the method and details necessary to calculate and implement retry back-off. The consumers shouldn't have access to this calculate method because they shouldn't need to call it. This could instead be moved to the common request implementation.